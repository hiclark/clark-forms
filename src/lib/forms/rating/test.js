import React from 'react';

import Rating from './';
import Check from '../../../demo/assets/icons/check.svg';

const RATING = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
];

const RATING_KEY = { left: 'Left', right: 'Right' };

describe('<Rating />', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      name: 'rating',
      values: RATING,
      ratingKey: RATING_KEY,
      icon: <Check />,
    };
    wrapper = shallow(<Rating {...props} />);
  });

  describe('<Rating />', () => {
    test('renders correctly', () => {
      expect(wrapper).toHaveLength(1);
    });

    test('render name prop', () => {
      expect(wrapper.prop('name')).toMatch('rating');
    });

    test('render values prop', () => {
      expect(wrapper.prop('values')).toEqual(RATING);
    });

    test('render ratingKey prop', () => {
      expect(wrapper.prop('ratingKey')).toEqual(RATING_KEY);
    });

    test('render icon prop', () => {
      expect(wrapper.prop('icon')).toEqual(<Check />);
    });
  });

  test('snapshot test <Rating />', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
