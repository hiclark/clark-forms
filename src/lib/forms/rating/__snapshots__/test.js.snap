// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> matches the snapshot 1`] = `
<Field
  component={[Function]}
  icon={
    <Check
      viewBox="0 0 14 10"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  name="rating"
  ratingKey={
    Object {
      "left": "Left",
      "right": "Right",
    }
  }
  values={
    Array [
      Object {
        "label": "1",
        "value": 1,
      },
      Object {
        "label": "2",
        "value": 2,
      },
      Object {
        "label": "3",
        "value": 3,
      },
      Object {
        "label": "4",
        "value": 4,
      },
      Object {
        "label": "5",
        "value": 5,
      },
    ]
  }
/>
`;
